features/FEATURE_LWIP/lwip-interface/lwip_stack.c

```diff
@@ -293,7 +293,9 @@ static int get_ip_addr_type(const ip_addr_t *ip_addr)
         return IPADDR_TYPE_V4;
     }
 #endif
+#if !LWIP_IPV6 && !LWIP_IPV4
```

Comment:
Should this be "#if LWIP_IPV6 && LWIP_IPV4"

Reply:
As in this?
```c++
#if LWIP_IPV6 && LWIP_IPV4
    return IP_ADDR_ANY;
#if LWIP_IPV4
....
#if LWIP_IPV6
...
```
I'm honestly not sure what the original attention was, the way it was written seemed to be only return IP_ADDR_ANY if neither LWIP_IPV4 or LWIP_IPV6 was enabled, but the name IP_ADDR_ANY sounds like it should be if both are enabled.

Reply:
IP_IS_V6/IP_IS_V4 defines do something (other than evaluate to true) only when both LWIP_IPV6 and LWIP_IPV4 flags are enabled. Also disabling both LWIP_IPV6 and LWIP_IPV4 is not valid configuration and will return compilation error. So, the function can reach the return at the end only in dual ipv4v6 configuration. 

Currently get_ip_addr_type() function is used by add dns address function and it is not made to be fully general utility function. That is because the dns function calling it never uses it to evaluate address that type is any, and is interested only whether address is either ipv4 or ipv6.

If function would be made more general, there could be also evolution under LWIP_IPV6 && LWIP_IPV4 flags to return IP_ADDR_ANY only in case address type really is any.  And to return -1 or some other error, in case address type is not valid at all (field is somehow corrupted).

But I think that since currently this function is used only for purposes of add dns address, adding the above evaluations are not needed and flagging the return with LWIP_IPV6 && LWIP_IPV4 flags is enough.

Reply:
At this point, I'd be inclined to put a pragma in to shut the compiler up. The ifdefs already in that function are excessive - presumably they're to silence a "constant expression in condition"?

Given the form of the lwIP dual IPv4/v6 code, silencing "unreachable code" and "constant expression" would allow you to just freely use those IS_V4/IS_V6 macros without it getting in the way.

(The Linux kernel is built with those warnings disabled, because there are a lot of similar macros that can either be constant or variable depending on config switches).

In libservice's ns_types.h, you can see NS_DUMMY_DEFINITIONS_OK, which attempts to disable some of them...

Reply:
Using pragma is ok for me. Yes, I agree that it is not very good that we have these IP_IS_V6/IP_IS_V4 etc. defines that would allow to write clear code that works with all configurations, but because of compiler warnings we need to use ifdefs also when using macros and not just when defining them.